typings/objection/index.d.ts

```diff
@@ -183,8 +183,23 @@ declare namespace Objection {
     [relationName: string]: RelationMapping;
   }
 
+  interface RelationProperty {
+    size: number;
+    modelClass: Model;
+    props: string[];
+    cols: string[];
+  }
+
   interface Relation {
-    // TODO should this be something other than a tagging interface?
+    name: string;
+    ownerModelClass: Model;
+    relatedModelClass: Model;
```

Comment:
There are not actually `Model` instances but `Model` constructors, so you should use `ModelClass<any>` here.